integrations/mistral/src/haystack_integrations/components/converters/mistral/ocr_document_converter.py

```diff
@@ -0,0 +1,367 @@
+    - Annotations: https://docs.mistral.ai/capabilities/document_ai/annotations/
+
+    **Usage Example:**
+    ```python
+    from haystack.utils import Secret
+    from haystack_integrations.mistral import MistralOCRDocumentConverter
+    from mistralai.models import DocumentURLChunk, ImageURLChunk, FileChunk
+
+    converter = MistralOCRDocumentConverter(
+        api_key=Secret.from_env_var("MISTRAL_API_KEY"),
+        model="mistral-ocr-2505"
+    )
+
+    # Process multiple sources
+    sources = [
+        DocumentURLChunk(document_url="https://example.com/document.pdf"),
+        ImageURLChunk(image_url="https://example.com/receipt.jpg"),
+        FileChunk(file_id="file-abc123"),
+    ]
+    result = converter.run(sources=sources)
+
+    documents = result["documents"]  # List of 3 Documents
+    raw_responses = result["raw_mistral_response"]  # List of 3 raw responses
+    ```
+
+    **Structured Output Example:**
+    ```python
+    from pydantic import BaseModel, Field
+    from haystack_integrations.mistral import MistralOCRDocumentConverter
+
+    # Define schema for structured image annotations
+    class ImageAnnotation(BaseModel):
+        image_type: str = Field(..., description="The type of image content")
+        short_description: str = Field(..., description="Short natural-language description")
+        summary: str = Field(..., description="Detailed summary of the image content")
+
+    # Define schema for structured document annotations
+    class DocumentAnnotation(BaseModel):
+        language: str = Field(..., description="Primary language of the document")
+        chapter_titles: List[str] = Field(..., description="Detected chapter or section titles")
+        urls: List[str] = Field(..., description="URLs found in the text")
+
+    converter = MistralOCRDocumentConverter(
+        api_key=Secret.from_env_var("MISTRAL_API_KEY"),
+        model="mistral-ocr-2505",
+    )
+
+    sources = [DocumentURLChunk(document_url="https://example.com/report.pdf")]
+    result = converter.run(
+        sources=sources,
+        bbox_annotation_schema=ImageAnnotation,
+        document_annotation_schema=DocumentAnnotation,
+    )
+
+    documents = result["documents"]
+    raw_responses = result["raw_mistral_response"]
+    ```
+    """
+
+    def __init__(
+        self,
+        api_key: Secret = Secret.from_env_var("MISTRAL_API_KEY"),
+        model: str = "mistral-ocr-2505",
+        include_image_base64: bool = False,
+        pages: Optional[List[int]] = None,
+        image_limit: Optional[int] = None,
+        image_min_size: Optional[int] = None,
+        cleanup_uploaded_files: bool = True,
+    ):
+        """
+        Creates a MistralOCRDocumentConverter component.
+
+        :param api_key:
+            The Mistral API key. Defaults to the MISTRAL_API_KEY environment variable.
+        :param model:
+            The OCR model to use. Default is "mistral-ocr-2505".
```

Comment:
Where the user can find the list of available models? If there's a link, let's add it here.